### PR TITLE
Compatibility for druntime#2411

### PR DESCRIPTION
**Pull request by thewilsonator**
Please do double-check the mangling.

### CODE REVIEW

**Comment by EyalIO** [comment]
@thewilsonator Thanks for the patch! Is the CI failure related to this change?

**Comment by thewilsonator** [comment]
The one in druntime#2411 is.

**Comment by thewilsonator** [comment]
The one here is because `core.thread` isn't imported.

**Comment by thewilsonator** [comment]
Green now.

**Comment by thewilsonator** [comment]
I think another tag is needed for dlangs build kite.

File: src/mecca/lib/memory.d [hunk 1]
```diff
@@ -441,8 +442,19 @@ struct DRuntimeStackDescriptor {
...
+    static if (__traits(hasMember, Thread, "_locks")) {
+        pragma(mangle, "_D4core6thread6Thread6_locksG2G72v") extern __gshared static
             void[__traits(classInstanceSize, Mutex)][2] _locks;
+        @notrace private Mutex _slock() nothrow @nogc {
+            return cast(Mutex)_locks[0].ptr;
+        }
+    } else {
+        pragma(mangle,"_D4core6thread6Thread6_slockG72v") extern __gshared static
+            void[__traits(classInstanceSize, Mutex)] _slock;
+        @notrace private Mutex _slock() nothrow @nogc {
```
Comment (jacob-carlborg) [inline src/mecca/lib/memory.d:454]: @thewilsonator How does this work? Aren't there two members with the same name (`_slock`) within the same scope?

**Pull request merged**